rules/k8s/logging-elasticsearch.rules.yaml

```diff
@@ -0,0 +1,190 @@
+groups:
+- name: logging-elasticsearch
+  rules:
+
+  # Recording rules
+  # #####################
+
+  # Bulk requests rates
+  # =====================
+  - record: bulk:rejected_requests:rate2m
+    expr: rate(es_threadpool_threads_count{name="bulk", type="rejected"}[2m])
+
+  - record: bulk:completed_requests:rate2m
+    expr: rate(es_threadpool_threads_count{name="bulk", type="completed"}[2m])
+
+  # If there are no bulk rejections then we get can 0/0 which is NaN. Although this might seem counterintuitive
+  # it is in fact valid result wrt to Prometheus and a good practice also, see:
+  # https://stackoverflow.com/questions/47056557/how-to-gracefully-avoid-divide-by-zero-in-prometheus
+  - record: bulk:reject_ratio:rate2m
+    expr: sum by (cluster, instance, node) (bulk:rejected_requests:rate2m) / on (cluster, instance, node) (bulk:completed_requests:rate2m)
+
+  # Alerting rules:
+  # #####################
+
+  # Cluster health alerts
+  # =====================
+  #
+  # Cluster health can become RED by natural cause or by critical cause.
+  #
+  # The natural causes do not last long and they include:
+  #   - a new index is created and all its primary shards haven’t been allocated yet
+  #   - a master node has not been elected yet (for example master node dies and a new master node hasn’t been
+  #     elected yet, and this can take some time if the cluster load is high...)
+  #
+  # If the natural cause takes long or the cause is different than those listed above then it is considered
+  # the critical cause.
+  #
+  # Cluster health becomes YELLOW if not all index replica shards are allocated. If the goal is to have GREEN cluster
+  # health (ie number of replica shards is configured accordingly) but the status stays YELLOW for too long then this
+  # is considered serious.
+  #
+  - alert: Cluster_Health_Status_RED
```

Comment:
Severity of alerts is encoded in their labels, not alert name. Alerts are camel case and should describe the alert precisely, more something like "ElasticSearchClusterNotHealthy".

Reply:
just a convention, I personally couldn't care less but prefer consistency